### ISSUE TITLE
When using camelcase rule, can i just ignore property access and still check property definitions?

### ISSUE THREAD

**Issue by BiosSun** [eslint/eslint#13895]
**What rule do you want to change?**

[Require CamelCase (camelcase)](https://eslint.org/docs/rules/camelcase)

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Add a new value to `properties` option, such as `define`.

**Please provide some example code that this change will affect:**

```js
/*eslint camelcase: ["error", {properties: "define"}]*/

// error
var obj = { my_pref: 1 };

// ignore
obj.my_pref;
```

**Comment by mdjermanovic** [comment]
Hi @BiosSun, thanks for the issue!

Doesn't `"properties": "always"` (default) already cover this?

```js
/* eslint camelcase: ["error", { "properties": "always" }] */

var obj = { my_pref: 1 }; // error

obj.my_pref = foo; // error

var foo = obj.my_pref; // ok
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhbWVsY2FzZTogW1wiZXJyb3JcIiwgeyBcInByb3BlcnRpZXNcIjogXCJhbHdheXNcIiB9XSAqL1xuXG52YXIgb2JqID0geyBteV9wcmVmOiAxIH07IC8vIGVycm9yXG5cbm9iai5teV9wcmVmID0gZm9vOyAvLyBlcnJvclxuXG52YXIgZm9vID0gb2JqLm15X3ByZWY7IC8vIG9rXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

It allows reading from underscored properties, but not creating new ones (incl. assigning to, since it can create a new property).
